### PR TITLE
Fixes issue with missing authors on commits

### PR DESCRIPTION
**Pull request by perploug**
Due to data inconsistencies on the github api side, some commits are missing an author object, meaning the check fails silently.

This adds a check for commits without author and filters them out in the comparison process.

Fixes: #29 

Signed-off-by: Per Ploug <per.ploug@zalando.de>

### CODE REVIEW

**Comment by perploug** [comment]
👍

**Comment by perploug** [comment]
Hey @fokusferit - if you have time, it is a minor check added to the commit comparison to fix an edge-case found in Kopf (test case added)

File: src/tasks/approvals.ts [hunk 1]
```diff
@@ -92,14 +93,17 @@ export default class FourEyePrincipleTask extends BaseTask<IApprovalsConfig> {
       number: context.payload.pull_request.number
     });*/
 
+    response.data.commits as ReposCompareCommitsResponse
```
[the remark this answers is not in the document]
Comment (perploug) [reply src/tasks/approvals.ts]: Go home, you are drunk

**Comment by perploug** [comment]
👍

**Comment by fokusferit** [comment]
Due to holidays I just saw this today.

File: src/tasks/approvals.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fokusferit) [reply src/tasks/approvals.ts]: Do you need to this for TS? or could you just change line 99 into

`return (response.data.commits as ReposCompareCommitsResponse)` ?

File: src/zincr.ts [hunk 2]
```diff
@@ -31,23 +31,33 @@ export class Zincr {
...
+
+
    
     // if there is no pull request or the state is not open, no reason to continue
     if (!pullRequest || pullRequest.state !== "open") return;
 
+    // only trigger on valid PR events
+    if (
+      event == "pull_request" &&
+      valid_pr_actions.indexOf(context.payload.action) < 0
```
Comment (fokusferit) [inline src/zincr.ts]: Doesn't this mean "if(Event is PR && the Action in the Payload is not a Valid PR Action)" ? Seems different then the comment of "only trigger on valid PR Events". Shouldn't this be ` valid_pr_actions.indexOf(context.payload.action) >= 0 `

Comment (fokusferit) [reply src/zincr.ts]: Ignore, I see you don't return stuff, so the code comment is just a bit confusing.

**Comment by fokusferit** [comment]
👍
